custom_components/ha_washdata/cycle_detector.py

```
@@ -494,6 +522,106 @@ def _sanitize_expected_duration(
+            cut = 0
+            for i in range(1, len(window)):
+                if (window[i][0] - window[i - 1][0]).total_seconds() > max_gap:
+                    cut = i
+            window = window[cut:]
+        if len(window) < SMART_TERM_TAIL_MIN_POINTS:
+            return None
+        span = (window[-1][0] - window[0][0]).total_seconds()
+        if span <= 0:
+            return None
+        energy = 0.0
+        for (t0, p0), (t1, p1) in zip(window, window[1:]):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)